t/frame.c

```diff
@@ -89,6 +89,34 @@ static void test_ack_decode(void)
         ok(decoded.smallest_acknowledged == 0x34 - 1 - 2 - 3 - 4 - 5 + 1);
     }
 
+    { /* Bogus ACK Frame larger than the internal buffer */
+        uint8_t pat[1024], *pos = pat;
+        const uint8_t *src = pat;
+        int i, range_sum;
+        quicly_ack_frame_t decoded;
+        pos = quicly_encodev(pos, 0xFA00);
+        pos = quicly_encodev(pos, 0);
+        pos = quicly_encodev(pos, QUICLY_ACK_MAX_GAPS + 1);
+        pos = quicly_encodev(pos, 8);
+        for (i = 0; i <= QUICLY_ACK_MAX_GAPS ; ++i) {
+            pos = quicly_encodev(pos, i); // gap
+            pos = quicly_encodev(pos, i % 10); // ack-range
+        }
+
+        ok(quicly_decode_ack_frame(&src, pos, &decoded, 0) == 0);
+        ok(decoded.largest_acknowledged == 0xFA00);
+        ok(decoded.ack_delay == 0);
+        ok(decoded.num_gaps == QUICLY_ACK_MAX_GAPS);
+        ok(decoded.ack_block_lengths[0] == 8 + 1); // first ack-range
+        range_sum = decoded.ack_block_lengths[0];
+        for (i = 0; i < QUICLY_ACK_MAX_GAPS; ++i) {
+            ok(decoded.gaps[i] == i + 1);
+            ok(decoded.ack_block_lengths[i + 1] == (i % 10) + 1);
+            range_sum += decoded.gaps[i] + decoded.ack_block_lengths[i + 1];
+        }
+        ok(decoded.smallest_acknowledged == 0xFA00 - range_sum + 1);
```

Comment:
I think checking if `src` is at the expected position is a good way to see if we have the problem stated above.

Reply:
Good point. I'll add a check to test that the  `src` pointer was forwarded to the expected end of the raw frame.